youtube_dl/extractor/dailymotion.py

```diff
@@ -390,12 +395,19 @@ def _real_extract(self, url):
         video_info = self._parse_json(self._search_regex(
             r'var\s+info\s*=\s*([^;]+);', webpage, 'video info'), video_id)
 
-        # TODO: parse ios_url, which is in fact a manifest
-        video_url = video_info['mp4_url']
+        is_live = video_info['mode'] == 'live'
+
+        formats = self._extract_m3u8_formats(self.redirect(video_info['ios_url'], video_id), video_id)
+
+        if is_live:
+            title = self._live_title(title)
+        else:
+            formats.append({'url': self.redirect(video_info['mp4_url'], video_id), 'format_id': 'mp4'})
```

Comment:
Apparently those urls use a normal HTTP 302 redirect, since the user may not download this format we can avoid the additional request. I haven't tested the m3u8 url, but if it also uses a normal redirect we should also avoid it.